inputvalidation/InputValidation.java

```
@@ -0,0 +1,504 @@
+            throw new RuntimeException(e.getMessage(), e);
+        }
+    }
+
+    private void updateResource(String customerId, Arn arn, Map<String, String> addedTags, List<String> removedTags, Long internalId) {
+        try {
+            taggingHelper.updateResource(customerId, arn, addedTags, removedTags, internalId);
+        } catch (RDSTaggingInvalidParameterException e) {
+            throw new InvalidParameterValueException(e.getMessage(), e);
+        } catch (RDSTaggingAccessDeniedException e) {
+            throw new InvalidParameterCombinationException(e.getMessage(), e);
+        } catch (RDSTaggingConcurrentAccessException e) {
+            throw new InvalidParameterCombinationException(e.getMessage(), e);
+        } catch (RDSTaggingInternalIdMismatchException e) {
+            throw new InvalidParameterCombinationException(e.getMessage(), e);
+        } catch (RDSTaggingTooManyTagsException e) {
+            throw new InvalidParameterCombinationException(e.getMessage(), e);
+        } catch (Exception e) {
+            log().warn("Exception calling updateResource", e);
+            throw new RuntimeException(e.getMessage(), e);
+        }
+    }
+
+    @Override
+    public void deleteResource(String customerId, Arn arn) {
+        try {
+            taggingHelper.deleteResource(customerId, arn);
+        } catch (RDSTaggingException e) {
+            log().warn("Exception calling deleteResource", e);
+        }
+    }
+
+    private Long getInternalId(Arn arn) {
+        TaggableResource resource = validateResourceExistenceAndRetrieveResource(arn);
+        return getInternalId(arn, resource);
+    }
+
+    private Long getInternalId(Arn arn, TaggableResource resource) {
+        return resource.getInternalId();
+    }
+
+    private void validateResourceName(Arn arn) {
+        try {
+            validateResourceName(arn.toString());
+        }
+        catch (Exception e) {
```

Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**: While wrapping the caught exception into a custom one, information about the caught exception is being lost, including information about the stack trace of the exception.

**Fix**: If the caught exception object does not contain sensitive information, consider passing it as the &#34;rootCause&#34; or inner exception parameter to the constructor of the new exception before throwing the new exception. (Note that not all exception constructors support inner exceptions. Use a wrapper exception that supports inner exceptions.) 
[Learn more](https://www.ibm.com/support/pages/best-practice-catching-and-re-throwing-java-exceptions)